### ISSUE TITLE
PEP 3105 Make print a function in twisted/mail

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8403]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8403|
|Type|enhancement|
|Created|2016-06-03 10:48:25Z|
|Branch|https://github.com/twisted/twisted/tree/8403-rodrigc-print|

This ticket was split out of [ticket:5812] which was too big to review.

PEP 3105 ( ​​​https://www.python.org/dev/peps/pep-3105/ ) **print** deprecates has changed from a statement to a function.

In Python 2.6 and Python 2.7, the default is print as a statement

```
print "Hello"
```

In Python 3.x, the above example is an error, since only print as a function is available.

To consistently use **print** as a function
on Python 2.6, Python 2.7, and Python 3.x, this should be done:

```
from __future__ import print_function

print("Hello")
```

<details><summary>Searchable metadata</summary>

```
trac-id__8403 8403
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8403_rodrigc_print 8403-rodrigc-print
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__None None
time__1464950905841358 1464950905841358
changetime__1465746910420697 1465746910420697
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/119

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Thanks for the patch.

The usage of print outside of the scrips should be converted into something else

The usage in twisted/mail/test/pop3testserver.py is ok.

In  twisted/mail/test/test_pop3.py the parts should be part of the assertion failure message.

Please consider my comments and resubmit on top of your existing branch (no rebase) :)

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @adiroiban|
|-|-|

See:
https://github.com/twisted/twisted/pull/119

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set owner to @hawkowl|
|-|-|

Looks good, just waiting for builders to go green...

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset eba908ee8581d904ec019680787c121ccf80a435
```
#!CommitTicketReference repository="" revision="eba908ee8581d904ec019680787c121ccf80a435"
Merge 8403-rodrigc-print: Make print a function in twisted/mail

Author: rodrigc
Reviewers: adiroiban, hawkowl
Fixes: #8403
```
